File: Curso Logica de Programação Neri Neitzke/Algoritmos em Java/primo_logico.java
// Prof Neri Aldoir Neitzke   ULBRA
// www.INFORMATICON.com.br - videoaulasneri@gmail.com
/*Problema: Crie um algoritmo que leia um numero inteiro, e diga se ele é um numero primo ou não.*/
import javax.swing.JOptionPane;
class primo_logico
{
   public static void main(String args[])
   {
       int numero,i;
       boolean contador = true;
       i = 2;
       numero = Integer.parseInt(JOptionPane.showInputDialog("Digite um número .: "));
       while((i <= (numero/2)) && (contador == true))
       {
          System.out.println(i);
          if (numero % i == 0)
             contador = false;
          i = i+1;
       }
       if (contador) //(contador == true)
           JOptionPane.showMessageDialog(null,"O numero "+numero+" e primo");
       else
           JOptionPane.showMessageDialog(null,"O numero "+numero+" nao e primo");

   }
}

File: Curso Logica de Programação Neri Neitzke/Algoritmos em Pascal/pascal/primo_logico.pas
// Prof Neri Aldoir Neitzke         - ULBRA
// www.INFORMATICON.com.br - videoaulasneri@gmail.com
{Problema: Crie um algoritmo que leia um numero inteiro, e diga se ele é um numero primo ou não. }
program primo;
Var numero,i : integer;
    num_primo : boolean; //lógico
begin
     num_primo := true;
     I := 2;
     write('Digite um numero .: ');
     readln(numero);
     while((i <=  (numero/2)) and (num_primo = true))  do
     begin
        writeln(i);
        if (numero mod i = 0) then
           num_primo := false;
        I := I + 1;
     end;
     if num_primo then       //if num_primo = true  then
           writeln('O numero ',numero,' e primo')
     else
           writeln('O numero ',numero,' nao e primo');

    writeln('..................Final do Sistema..................... ');
    readln();
end.
